src/components/Header/Header.js

```diff
@@ -25,6 +25,29 @@ const SearchEngine = () => {
 
 };
 
+const ShoppingCart = () => {
+
+    const navigate = useNavigate();
+    const { cartItems } = useContext(CartItemsContext);
+    const [totalItems, setTotalItems] = useState(0);
+    useEffect(()=>
+        {
+            let tempTotalItems=0;
+            cartItems.forEach((e)=>{tempTotalItems+=e.quantity })
```

Comment:
You could move this logic to the Context so you don't need to do this in every component that you use.